projects/fhi-angular-highcharts/src/lib/fhi-diagram-type-navs/fhi-diagram-type-nav.component.html

```diff
@@ -1,38 +1,45 @@
-<div class="fhi-diagram-type-nav mb-3">
-  <div class="btn-group fhi-btn-group mb-1" role="group" aria-label="Visningsvalg">
-    <ng-container *ngFor="let navElement of navMenu; index as i">
+<div class="fhi-diagram-type-nav mb-3" *ngIf="showNav">
+  <div class="btn-group fhi-btn-group mb-1"
+       role="group"
+       aria-label="Velg tabell, kart eller graf">
+
+    <ng-container *ngFor="let group of navDiagramTypeGroups">
       <input class="btn-check fhi-btn-group__check"
               type="radio"
-              name="visualisationNav"
-              aria-controls="chartTypeNav"
-              [disabled]="navElement.icon === 'geo-alt'"
-              [id]="'navElm' + i"
-              [checked]="i === 0"
-              [attr.aria-haspopup]="navElement.hasChartTypes"
-              (change)="chartIsActive = navElement.hasChartTypes">
-      <label [for]="'navElm' + i" class="btn fhi-btn-group__btn"
-             [ngClass]="{ 'disabled': navElement.icon === 'geo-alt' }"
-             (click)="changeChartType(navElement.icon !== null ? navElement.icon : currentChartType)">
-        <i class="icon-md" [ngClass]="navElement.icon !== null ? 'icon-' + navElement.icon : 'icon-' + currentChartType"></i>
-        {{ navElement.name }}
+              name="{{ 'diagram-type-nav' + navId }}"
+              id="{{ group.id + navId }}"
+              aria-controls="diagram-type-children-nav"
+              [attr.aria-haspopup]="diagramTypeGroupIsChart(group.id)"
+              [disabled]="group.isDisabled"
+              [checked]="group.id === currentDiagramTypeGroup"
+              (change)="navigate(group.diagramType)">
+
+      <label class="btn fhi-btn-group__btn"
+             for="{{ group.id + navId }}"
+             [ngClass]="{ 'disabled': group.isDisabled }">
+        <i class="icon-md" [ngClass]="'icon-' + group.icon"></i>
```

Comment:
I don't know exactly why, but on the examples the map menu option shows a 'graph-up' icon